docs/api/utility.rst

```diff
@@ -0,0 +1,325 @@
+.. # Copyright (c) 2017-2019, Lawrence Livermore National Security, LLC and
+.. # other BLT Project Developers. See the top-level COPYRIGHT file for details
+.. # 
+.. # SPDX-License-Identifier: (BSD-3-Clause)
+
+Utility Macros
+==============
+
+
+General Utility Macros
+----------------------
+
+
+blt_list_append
+~~~~~~~~~~~~~~~
+
+.. code-block:: cmake
+
+    blt_list_append( TO <list> ELEMENTS [ <element>...] IF <bool> )
+
+Appends elements to a list if the specified bool evaluates to true.
+
+This macro is essentially a wrapper around CMake's `list(APPEND ...)`
+command which allows inlining a conditional check within the same call
+for clarity and convenience.
+
+This macro requires specifying:
+
+    * The target list to append to by passing TO <list>
```

Comment:
I think the API docs would be clearer if the arguments were given as a "definition list" instead of a bulleted list.

See: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html?highlight=definition%20list

Reply:
I agree.. though I'm going to leave that until a follow up PR which restructures a lot of the content.